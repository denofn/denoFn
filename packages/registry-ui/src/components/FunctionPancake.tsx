import React from "react";

import { Pancake } from "./Pancake";
import { StatusBadge } from "./StatusBadge";
import { ViewsContext } from "./Views";
import { WordWrap } from "./WordWrap";

type FunctionPancakeProps = {
  f: string;
  status: boolean;
};

export function FunctionPancake({ f, status }: FunctionPancakeProps) {
  const [, dispatch] = React.useContext(ViewsContext);

  return (
    <Pancake
      onClick={() =>
        dispatch({
          current: "manage",
          manage: f,
        })
      }
      key={f}
    >
      <WordWrap>{f}</WordWrap>
      <StatusBadge status={status} statusText={["warm", "cold"]} />
    </Pancake>
  );
}
