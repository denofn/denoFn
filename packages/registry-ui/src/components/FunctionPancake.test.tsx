import { render, cleanup } from "@testing-library/react";
import React from "react";

import { FunctionPancake } from "./FunctionPancake";

afterEach(cleanup);

it("renders with the correct styles", () => {
  const { asFragment } = render(<FunctionPancake f="testFn" status />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders COLD status", () => {
  const { asFragment } = render(<FunctionPancake f="testFn" status={false} />);
  expect(asFragment()).toMatchSnapshot();
});
