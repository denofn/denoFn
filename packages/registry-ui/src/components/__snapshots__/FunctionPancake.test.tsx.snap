// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders COLD status 1`] = `
<DocumentFragment>
  <div
    class="_e0dnmk _zjik7 _5dh3r6 _qdeacm _2o6p8u _1qesdsb _1khn195 _sy4y6p _1x4v58x _1dvcxr3 _u68uts"
  >
    <span
      class="_1m5hyg0"
    >
      testFn
    </span>
    <div
      class="_iabeq1 _1r5gb7q _j033lg _1efi8gv _180q8om _1dbsnr2 _r24itg _50zrmy _6zs611 _1y5pc60"
    >
      cold
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders with the correct styles 1`] = `
<DocumentFragment>
  <div
    class="_e0dnmk _zjik7 _5dh3r6 _qdeacm _2o6p8u _1qesdsb _1khn195 _sy4y6p _1x4v58x _1dvcxr3 _u68uts"
  >
    <span
      class="_1m5hyg0"
    >
      testFn
    </span>
    <div
      class="_iabeq1 _1r5gb7q _j033lg _1efi8gv _180q8om _1dbsnr2 _r24itg _50zrmy _st1wxx _1y5pc60"
    >
      warm
    </div>
  </div>
</DocumentFragment>
`;
